Services/Mail/classes/AutoResponder/AutoResponderArrayCollection.php

```diff
@@ -0,0 +1,172 @@
+<?php declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+namespace ILIAS\Services\Mail\AutoResponder;
+
+use InvalidArgumentException;
+use ArrayIterator;
+
+class AutoResponderArrayCollection implements AutoResponderCollection
```

Comment:
Could we try to use the `Generics` doc syntax from `PHPStan`? https://phpstan.org/blog/generics-by-examples#define-a-generic-interface